packages/medical-domain/src/domains/repositories/medical/datum/alarm-event.service.ts

```diff
@@ -0,0 +1,123 @@
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import MedicalDataOptions from '../../../models/medical/medical-data-options.model';
+import { AlarmEvent } from '../../../models/medical/datum/alarm-event.model';
+import BaseDatumService from './basics/base-datum.service';
+import DurationService from './basics/duration.service';
+import { DatumType } from '../../../models/medical/datum/enums/datum-type.enum';
+import { DeviceEventSubtype } from '../../../models/medical/datum/enums/device-event-subtype.enum';
+import { DatumProcessor } from '../../../models/medical/datum.model';
+import { AlarmEventType } from '../../../models/medical/datum/enums/alarm-event-type.enum';
+import { AlarmLevel } from '../../../models/medical/datum/enums/alarm-level.enum';
+import DatumService from '../datum.service';
+import { defaultWeekDaysFilter, WeekDaysFilter } from '../../../models/time/date-filter.model';
+import { AlarmCode } from '../../../models/medical/datum/enums/alarm-code.enum'
+
+const CATEGORIZED_ALARM_CODES = {
+  DEVICE: {
+    ALARM: [
+      AlarmCode.EmptyPumpBattery,
+      AlarmCode.EmptyInsulinCartridge,
+      AlarmCode.InsulinCartridgeExpired,
+      AlarmCode.Occlusion,
+      AlarmCode.SensorSessionExpired
+    ]
+  },
+  HYPOGLYCEMIA: {
+    ALARM: [AlarmCode.Hypoglycemia, AlarmCode.LongHypoglycemia],
+    ALERT: [AlarmCode.UrgentLowSoon, AlarmCode.NoReadingsHypoglycemiaRisk]
+  },
+  HYPERGLYCEMIA: {
+    ALARM: [AlarmCode.LongHyperglycemia],
+    ALERT: [AlarmCode.Hyperglycemia, AlarmCode.SuddenRiseInGlycemia]
+  }
+}
+
+const getAlarmEventType = (alarmCode: AlarmCode, alarmLevel: string): AlarmEventType => {
+  const isHypoglycemiaAlarm = alarmLevel === AlarmLevel.Alarm && CATEGORIZED_ALARM_CODES.HYPOGLYCEMIA.ALARM.includes(alarmCode)
+  const isHypoglycemiaAlert = alarmLevel === AlarmLevel.Alert && CATEGORIZED_ALARM_CODES.HYPOGLYCEMIA.ALERT.includes(alarmCode)
+  const isHyperglycemiaAlarm = alarmLevel === AlarmLevel.Alarm && CATEGORIZED_ALARM_CODES.HYPERGLYCEMIA.ALARM.includes(alarmCode)
+  const isHyperglycemiaAlert = alarmLevel === AlarmLevel.Alert && CATEGORIZED_ALARM_CODES.HYPERGLYCEMIA.ALERT.includes(alarmCode)
+  const isDeviceAlarm = alarmLevel === AlarmLevel.Alarm && CATEGORIZED_ALARM_CODES.DEVICE.ALARM.includes(alarmCode)
+
+  if (isHypoglycemiaAlarm || isHypoglycemiaAlert) {
+    return AlarmEventType.Hypoglycemia
+  }
+
+  if (isHyperglycemiaAlarm || isHyperglycemiaAlert) {
+    return AlarmEventType.Hyperglycemia
+  }
+
+  if (isDeviceAlarm) {
+    return AlarmEventType.Device
+  }
+
+  return AlarmEventType.Unknown
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I find it unusual to compute all `if` conditions before like that. `isDeviceAlarm` is always computed even though it the last condition that is checked.
I see no issue for this kind of computation. This is still understandable. Performances wise, the impact is probably extremely small. However this kind of pattern has to scale badly (as it consist in computing things that are not being used) :thinking: 

Here is a suggestion that I would find better without breaking the way you approach the issue:

```suggestion
  const isHypoglycemiaAlarm = alarmLevel === AlarmLevel.Alarm && CATEGORIZED_ALARM_CODES.HYPOGLYCEMIA.ALARM.includes(alarmCode)
  const isHypoglycemiaAlert = alarmLevel === AlarmLevel.Alert && CATEGORIZED_ALARM_CODES.HYPOGLYCEMIA.ALERT.includes(alarmCode)

  if (isHypoglycemiaAlarm || isHypoglycemiaAlert) {
    return AlarmEventType.Hypoglycemia
  }

  const isHyperglycemiaAlarm = alarmLevel === AlarmLevel.Alarm && CATEGORIZED_ALARM_CODES.HYPERGLYCEMIA.ALARM.includes(alarmCode)
  const isHyperglycemiaAlert = alarmLevel === AlarmLevel.Alert && CATEGORIZED_ALARM_CODES.HYPERGLYCEMIA.ALERT.includes(alarmCode)
  
  if (isHyperglycemiaAlarm || isHyperglycemiaAlert) {
    return AlarmEventType.Hyperglycemia
  }


  const isDeviceAlarm = alarmLevel === AlarmLevel.Alarm && CATEGORIZED_ALARM_CODES.DEVICE.ALARM.includes(alarmCode)
  
  if (isDeviceAlarm) {
    return AlarmEventType.Device
  }

  return AlarmEventType.Unknown
```